app/src/main/res/values/strings.xml

```diff
@@ -11,16 +11,29 @@
     <!-- Title for dialog displayed on first start -->
     <string name="welcome_title">Welcome to Syncthing for Android</string>
 
-    <!-- Text for dialog displayed on first start -->
+    <!-- Welcome wizard -->
+    <!-- Slide 1 -->
+    <string name="introduction">Introduction</string>
     <string name="welcome_text">Syncthing is an open-source file synchronization application.\n\
 To share data with other devices, you need to add their unique device IDs to the device list. Afterwards you can select which folders to share with which devices.\n\
 Please report any problems you encounter via Github.</string>
 
-    <string name="cont">Continue</string>
+    <!-- Slide 2 -->
+    <string name="storage_permission_title">Storage permission</string>
+    <string name="storage_permission_desc">Syncthing requires the storage permission to do file synchronization. We ask you to grant the permission.</string>
+
+    <!-- Slide 3 -->
+    <string name="location_permission_title">Location permission</string>
+    <string name="location_permission_desc">Syncthing can be configured to run only on selected wifi networks. Due to android requirements, you need to grant the permission if you\'d like to use this feature. Otherwise, feel free to deny it.</string>
```

Comment:
Here the button makes sense, as it's voluntary. I'd turn the latter part around to focus on that choice instead of the android requirement. Something like `If you want to use this feature, press the button above to give the required location access to Syncthing. Otherwise you can skip this step.`

Reply:
I think this needs an explanation why...


> Syncthing can be configured to run only on selected wifi networks. Sadly, Android requires applications to have location permissions to be able look up active WiFi network name, as you can sometimes infer users location from the name of the network they are connected to. If you want to use this feature, press the button above to give the required location permissions to Syncthing. Otherwise you can skip this step.

Reply:
Hmm.. both suggestions are good. imsodin's is comforting the user but the user doesn't really get why a location permission should be granted for a wifi feature. Audrius' suggestion tells the truth (what for example complies with fdroid policy for app approval) but also looks a tiny bit scary as we say what nasty things we could do (but won't do :). For the first try on "later" public feedback I'll take Audrius' suggestion.

Reply:
Fine with me, but please drop the sadly. We want the user to be cheerful about using Syncthing ;)